components/script/dom/textcontrol.rs

```diff
@@ -0,0 +1,74 @@
+use dom::bindings::cell::DomRefCell;
+use dom::bindings::conversions::DerivedFrom;
+use dom::bindings::str::DOMString;
+use dom::event::{EventBubbles, EventCancelable};
+use dom::eventtarget::EventTarget;
+use dom::node::{Node, NodeDamage, window_from_node};
+use script_traits::ScriptToConstellationChan;
+use textinput::{SelectionDirection, TextInput};
+
+pub trait TextControl: DerivedFrom<EventTarget> + DerivedFrom<Node> {
+    fn textinput(&self) -> &DomRefCell<TextInput<ScriptToConstellationChan>>;
+
+    // https://html.spec.whatwg.org/multipage/#dom-textarea/input-selectionstart
+    fn dom_selection_start(&self) -> u32 {
+        self.textinput().borrow().get_selection_start()
```

Comment:
When I mentioned that whether the selection{Start, End, Range} functions matches the spec, I meant whether they are following the steps as defined in the spec. This function in particular does not seem to have included step 1 and 2.